### PR TITLE
UserSession class [1] - Add class and tests

### PR DESCRIPTION
**Pull request by nupplaphil**
waiting for #12020 :)

### CODE REVIEW

File: src/Core/Session/Model/SessionUsers.php [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+ *
+ */
+
+namespace Friendica\Core\Session\Model;
+
+use Friendica\Core\Session\Capability\IHandleSessions;
+use Friendica\Core\Session\Capability\IHandleSessionUsers;
+use Friendica\Model\Contact;
+
+class SessionUsers implements IHandleSessionUsers
```
Comment (MrPetovan) [inline src/Core/Session/Model/SessionUsers.php]: Why the plural in the class names? Additionally, I think I'd prefer `UserSession` as it makes more sense from an English language standpoint.

**Review by MrPetovan — CHANGES_REQUESTED** [review]

File: src/Core/Session/Model/UserSession.php [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+	{
+		if (!empty($this->session->get('authenticated')) && !empty($this->session->get('uid'))) {
+			return intval($this->session->get('uid'));
+		}
+
+		return false;
+	}
+
+	/** {@inheritDoc} */
+	public function getPublicContactId()
```
Comment (Quix0r) [inline src/Core/Session/Model/UserSession.php:51]: Can you please add proper returned type-hint, if possible? It feels like `: int` is missing here?

File: src/Core/Session/Model/UserSession.php [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+			}
+		} elseif (empty($this->session->get('authenticated'))) {
+			$this->publicContactId = false;
+		}
+
+		return $this->publicContactId;
+	}
+
+	/** {@inheritDoc} */
+	public function getRemoteUserId()
```
Comment (Quix0r) [inline src/Core/Session/Model/UserSession.php:69]: Same here?

File: src/DI.php [hunk 4]
```diff
@@ -219,6 +220,11 @@ public static function session()
 		return self::$dice->create(Core\Session\Capability\IHandleSessions::class);
 	}
 
+	public static function sessionUser(): IHandleUserSessions
```
Comment (Quix0r) [inline src/DI.php]: Cannot we rename this to `userSession` to reflect the same renaming for class `UserSession`?

File: static/dependencies.config.php [hunk 5]
```diff
@@ -224,6 +224,9 @@
 			['start', [], Dice::CHAIN_CALL],
 		],
 	],
+	\Friendica\Core\Session\Capability\IHandleUserSessions::class => [
```
Comment (Quix0r) [inline static/dependencies.config.php]: Hmm, I would choose to move this to the header of the file and "import" it with `use` instead.

**Review by Quix0r — CHANGES_REQUESTED** [review]

File: src/Core/Session/Model/UserSession.php (hunk 2 above)
Comment (nupplaphil) [reply src/Core/Session/Model/UserSession.php:51]: the typehint is not supported yet (`int|bool`)

File: src/Core/Session/Model/UserSession.php (hunk 3 above)
Comment (nupplaphil) [reply src/Core/Session/Model/UserSession.php:69]: the typehint is not supported yet (`int|bool`)

File: tests/src/Core/Session/UserSessionTest.php [hunk 6]
```diff
@@ -0,0 +1,218 @@
...
+	/**
+	 * @dataProvider dataGetRemoteUserId
+	 */
+	public function testGetRemoteUserId(array $data, $expected)
+	{
+		$userSession = new UserSession(new ArraySession($data));
+		$this->assertEquals($expected, $userSession->getRemoteUserId());
+	}
+
+	/// @fixme Add more data when Contact::getIdForUrl ist a dynamic class
```
Comment (MrPetovan) [inline tests/src/Core/Session/UserSessionTest.php]:
```suggestion
	/// @fixme Add more data when Contact::getIdForUrl is a dynamic class
```

**Review by MrPetovan — CHANGES_REQUESTED** [review]

**Review by MrPetovan — APPROVED** [review]

**Pull request merged**